src/Opdex.Platform.Application.Abstractions/Queries/VaultGovernances/Pledges/RetrieveVaultProposalPledgesWithFilterQuery.cs

```diff
@@ -0,0 +1,34 @@
+using MediatR;
+using Opdex.Platform.Common.Models;
+using Opdex.Platform.Domain.Models.VaultGovernances;
+using Opdex.Platform.Infrastructure.Abstractions.Data.Queries.VaultGovernances.Pledges;
+using System;
+using System.Collections.Generic;
+
+namespace Opdex.Platform.Application.Abstractions.Queries.VaultGovernances.Pledges;
+
+/// <summary>
+/// Request to retrieve pledges against a proposal in a vault.
+/// </summary>
+public class RetrieveVaultProposalPledgesWithFilterQuery : IRequest<IEnumerable<VaultProposalPledge>>
+{
+    /// <summary>
+    /// Creates a request to retrieve pledges against a proposal in a vault.
+    /// </summary>
+    /// <param name="vaultId">Id of the vault.</param>
+    /// <param name="proposalId">Non-public id of the proposal.</param>
+    /// <param name="cursor">Cursor filters.</param>
+    public RetrieveVaultProposalPledgesWithFilterQuery(ulong vaultId, ulong proposalId, VaultProposalPledgesCursor cursor)
```

Comment:
nit - if this is referring to the `proposalPublicId`, we might want to name it that way so its a bit clearer for developers.